Scripts/Generation_scripts/Grid.py

```diff
@@ -18,13 +18,33 @@
 import random
 
 class Grid():
-    def __init__(self):
-        self.location = 'Bahraich'
-        self.CLOVER_filepath = '/***YOUR LOCAL FILE PATH***/CLOVER 4.0'
+    def __init__(self,**kwargs):
+        # kwargs must include location
+        # kwargs can include grid_inputs (to override inputs in location files)
+        self.kwargs = kwargs
+        self.location = kwargs.get('location')
+        self.CLOVER_filepath = '.'
         self.location_filepath = self.CLOVER_filepath + '/Locations/' + self.location
         self.location_inputs = pd.read_csv(self.location_filepath + '/Location Data/Location inputs.csv',header=None,index_col=0)[1]
+        # Replace input values with keywords if specified
+        if kwargs.get('location_inputs'):
+            for i in kwargs.get('location_inputs'):
+                if not i[0] in self.location_inputs.index:
+                    print("Couldn't find entry",i[0],"in location_inputs. Perhaps it's misspelt in kwargs? Printing list of possible variables and exiting.")
```

Comment:
In general, python provides a nice way of formatting strings like this which are being printed out to the user. There are several different ways, but in the newest versions of python (3.7 onwards) you can use "f-strings":
```
>>> print(f"Couldn't find entry {i[0]} in location inputs.")
Couldn't find entry foo in location inputs.
```
In older versions of python, you could use the `.format()` attribute of a string to explicitly specify this:
```
>>> print("Couldn't find entry {} in location_inputs".format(i[0]})
Couldn't find entry foo in location_inputs.
```
When logging, and in really old python, you can use `%s` to say that it's a string that you're passing in there, though the first two methods are better 😄